File: src/main/java/org/appsugar/archetypes/repository/BaseJpaRepository.java
package org.appsugar.archetypes.repository;

import com.cosium.spring.data.jpa.entity.graph.domain.EntityGraph;
import com.cosium.spring.data.jpa.entity.graph.domain.EntityGraphUtils;
import com.cosium.spring.data.jpa.entity.graph.repository.EntityGraphQuerydslPredicateExecutor;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.repository.NoRepositoryBean;

import java.io.Serializable;


@NoRepositoryBean
public interface BaseJpaRepository<T, ID extends Serializable> extends JpaRepository<T, ID>, EntityGraphQuerydslPredicateExecutor {

    default EntityGraph fromPaths(String... attributePaths) {
        return EntityGraphUtils.fromAttributePaths(attributePaths);
    }
}

File: src/main/kotlin/org/appsugar/archetypes/repository/jpa/BaseJpaRepository.kt
package org.appsugar.archetypes.repository.jpa

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.querydsl.QuerydslPredicateExecutor
import org.springframework.data.repository.NoRepositoryBean
import java.io.Serializable


@NoRepositoryBean
interface BaseJpaRepository<T, ID : Serializable> : JpaRepository<T, ID>, QuerydslPredicateExecutor<T> {
}